src/solver/fold/unfold_backward.cpp

```diff
@@ -0,0 +1,166 @@
+                             const miopen::fold::UnfoldBwdProblemDescription& problem) const
+{
+    return true;
+}
+
+ConvSolution UnfoldBwd::GetSolution([[maybe_unused]] const ExecutionContext& context,
+                                    const miopen::fold::UnfoldBwdProblemDescription& problem) const
+{
+    std::ignore = context;
+    auto result = ConvSolution{miopenStatusSuccess};
+
+    auto in_dtype         = miopen::GetDataType(problem.GetDinputDesc().GetType());
+    auto dtype            = problem.GetDoutputDesc().GetType();
+    auto input_grad_dims  = problem.GetDinputDesc().GetLengths();
+    auto output_grad_dims = problem.GetDoutputDesc().GetLengths();
+
+    const int32_t N = static_cast<int32_t>(input_grad_dims[0]);
+    const int32_t C = static_cast<int32_t>(input_grad_dims[1]);
+    int32_t H       = static_cast<int32_t>(input_grad_dims[2]);
+    int32_t W       = static_cast<int32_t>(input_grad_dims[3]);
+
+    {
+        auto kernel        = KernelInfo{};
+        kernel.kernel_file = "MIOpenUnfold.cpp";
+        kernel.kernel_name = "UnfoldBackward4D";
+
+        const auto build_params = KernelBuildParameters{
+            {"MIOPEN_USE_FP16", static_cast<int>(dtype == miopenHalf)},
+            {"MIOPEN_USE_FP32", static_cast<int>(dtype == miopenFloat)},
+            {"MIOPEN_USE_FP64", static_cast<int>(dtype == miopenDouble)},
+            {"MIOPEN_USE_BFP16", static_cast<int>(dtype == miopenBFloat16)},
+        };
+        kernel.comp_options = build_params.GenerateFor(kbp::HIP{});
+
+        size_t xlocalsize = LOCAL_SIZE;
+        size_t xgridsize  = AlignUp(N * C * H * W, LOCAL_SIZE);
+        size_t ylocalsize = 1;
+        size_t ygridsize  = 1;
+        size_t zlocalsize = 1;
+        size_t zgridsize  = 1;
+        kernel.l_wk.push_back(xlocalsize);
+        kernel.l_wk.push_back(ylocalsize);
+        kernel.l_wk.push_back(zlocalsize);
+
+        kernel.g_wk.push_back(xgridsize);
+        kernel.g_wk.push_back(ygridsize);
+        kernel.g_wk.push_back(zgridsize);
+
+        result.construction_params.push_back(kernel);
+    }
+
+    result.invoker_factory = [](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle_, const AnyInvokeParams& raw_params) {
+            decltype(auto) kernel = handle_.Run(kernels.front());
+            decltype(auto) params = raw_params.CastTo<miopen::fold::InvokeParams>();
+
+            auto input_grad_tv    = get_inner_expanded_tv<4>(deref(params.dinputDesc));
+            auto output_grad_tv   = get_inner_expanded_tv<3>(deref(params.doutputDesc));
+            auto input_grad_dims  = deref(params.dinputDesc).GetLengths();
+            auto output_grad_dims = deref(params.doutputDesc).GetLengths();
+
+            int spatial_dim_size = input_grad_dims.size() - 2;
+            const int32_t N      = static_cast<int32_t>(input_grad_dims[0]);
+            const int32_t C      = static_cast<int32_t>(input_grad_dims[1]);
```

Comment:
no need to re-declare N, C

Reply:
resolve in commit [4f51b6e](https://github.com/moreh-dev/MIOpen/pull/37/commits/4f51b6e8b3f9adf009bf9d6f5a7999ee506ec908)